File: asyncSrc/emailVerification/logic.js
const {
  readApplication,
  sendVerificationEmail,
} = require('/opt/ports');

/**
 *
 * @param {string} applicationId Application ID
 * @param {string} email Email address to send verification token
 * @returns
 */
async function logic(applicationId, email, verificationToken) {
  const applicationData = await readApplication(applicationId);
  const {
    verificationUrl,
  } = applicationData;
  await sendVerificationEmail(email, verificationToken, verificationUrl);
}

module.exports = {
  logic,
};

File: gosrc/cmd/sendEmailVerification/logic.go
package main

import (
	"github.com/thomasstep/authentication-service/internal/adapters"
)

func logic(applicationId string, email string, token string) {
	// read application for verification url
	applicationItem, readAppErr := adapters.ReadApplication(applicationId)
	if readAppErr != nil {
		panic(readAppErr)
	}
	// send verification email
	adapters.SendVerificationEmail(email, token, applicationItem.VerificationUrl)
}
